### DISCUSSION TITLE
I would like to enable my abp blazor app to switch db provider at startup.

### DISCUSSION THREAD

**Discussion by Anders2020** [abpframework/abp#18707] [General] [answered]
I want my app to able to work with either sqlserver or sqlite or even mysql. I want to add a config setting to specify the provider and also change the connection string. The MainModule should then be able to wire up the correct dbcontext but still be using the same domain classes. Any thoughts?

**Comment by maliming** [comment]
hi

You can implement your `ConnectionStringResolver`.

I'm not sure if `DbContext` can switch providers at runtime.

https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.MultiTenancy/Volo/Abp/MultiTenancy/MultiTenantConnectionStringResolver.cs#L12

**Reply by Anders2020** [reply]
Thanks for your response.
 I dont mean using the same dbcontext, but having sqlserverdbcontext and sqlitedbcontext both inheriting from AbpDbContext<T>.  Only one dbcontext will be used for the application. The only difference is that the config will be used to specify which one to use.

I also want to know if in XX.EntityFrameworkCore.cs is it ok to have both DependsOn(typeof(AbpEntityFrameworkCoreSqlServerModule), typeof(AbpEntityFrameworkCoreSqliteModule), ...?

**Reply by maliming** [reply, marked as answer]
I think it is ok. You can give it a try.
